src/common/depthmap_toolkit/depthmap.py

```diff
@@ -175,41 +175,92 @@ def convert_2d_to_3d_oriented(self, sensor: int, x: float, y: float, depth: floa
                 point = self.convert_2d_to_3d_oriented(1, x, y, depth)
-                if abs(normal[1]) > 0.5:
-                    if abs(point[1] - floor) < 0.1:
-                        mask[x][y] = MASK_FLOOR
-
-        # highlight the focused child/object using seed algorithm
-        dirs = [[-1, 0], [1, 0], [0, -1], [0, 1]]
-        pixel = [int(self.width / 2), int(self.height / 2)]
-        stack = [pixel]
-        while len(stack) > 0:
-
-            # get a next pixel from the stack
-            pixel = stack.pop()
-            depth_center = self.parse_depth(pixel[0], pixel[1])
+                if abs(normal[1]) > 0.5 and abs(point[1] - floor) < 0.1:
+                    mask[x][y] = MASK_FLOOR
 
-            # add neighbor points (if there is no floor and they are connected)
-            if mask[pixel[0]][pixel[1]] == 0:
-                for direction in dirs:
-                    pixel_dir = [pixel[0] + direction[0], pixel[1] + direction[1]]
-                    depth_dir = self.parse_depth(pixel_dir[0], pixel_dir[1])
-                    if depth_dir > 0 and (depth_dir - depth_center) < 0.1:
-                        stack.append(pixel_dir)
+        return mask
 
-            # update the mask
-            mask[pixel[0]][pixel[1]] = MASK_CHILD
+    def detect_objects(self, floor: float) -> [np.array, list]:
 
-        return mask
+        # Detect objects/children using seed algorithm
+        current = -1
+        segments = []
+        dirs = [[-1, 0], [1, 0], [0, -1], [0, 1]]
+        mask = self.detect_floor(floor)
+        for x in range(self.width):
+            for y in range(self.height):
+                if mask[x][y] != 0:
+                    continue
+                pixel = [x, y]
+                aabb = [pixel[0], pixel[1], pixel[0], pixel[1]]
+                stack = [pixel]
+                while len(stack) > 0:
+
+                    # Get a next pixel from the stack
+                    pixel = stack.pop()
+                    depth_center = self.parse_depth(pixel[0], pixel[1])
+
+                    # Add neighbor points (if there is no floor and they are connected)
+                    if mask[pixel[0]][pixel[1]] == 0:
+                        for direction in dirs:
+                            pixel_dir = [pixel[0] + direction[0], pixel[1] + direction[1]]
+                            depth_dir = self.parse_depth(pixel_dir[0], pixel_dir[1])
+                            if depth_dir > 0 and abs(depth_dir - depth_center) < 0.1:
+                                stack.append(pixel_dir)
+
+                    # Update AABB
+                    if aabb[0] > pixel[0]:
+                        aabb[0] = pixel[0]
+                    if aabb[1] > pixel[1]:
+                        aabb[1] = pixel[1]
+                    if aabb[2] < pixel[0]:
+                        aabb[2] = pixel[0]
+                    if aabb[3] < pixel[1]:
+                        aabb[3] = pixel[1]
```

Comment:
You can use this shortcut, which does the same I think
```suggestion
                    aabb[3] = max(aabb[3], pixel[1])
```

Reply:
This code looks repetitive. Consider using numpy or a for-loop

Reply:
I think using for loop will only make code complex